pyomo/contrib/mindtpy/MindtPy.py

```diff
@@ -239,10 +242,15 @@ def solve(self, model, **kwds):
                     from_list=solve_data.best_solution_found.MindtPy_utils.variable_list,
                     to_list=MindtPy.variable_list,
                     config=config)
+                # MindtPy.variable_list does not include the unused variables, but component_data_objects(Var) does. component_data_objects(Var) should be used for both working_model and original_model.
+                solve_data.working_model.MindtPy_utils.deactivate()
+                if solve_data.working_model.find_component("_int_to_binary_reform") is not None:
+                    solve_data.working_model._int_to_binary_reform.deactivate()
                 copy_var_list_values(
-                    MindtPy.variable_list,
+                    [i for i in solve_data.working_model.component_data_objects(
+                        Var, active=True, descend_into=True) if not i.fixed],
                     [i for i in solve_data.original_model.component_data_objects(
-                        Var) if not i.fixed],
+                        Var, active=True, descend_into=True) if not i.fixed],
```

Comment:
This won't work if a Var that's used in an active Constraint or Objective is on a deactivated Block, which will be a problem as #1032 permeates. I think it would be better to use the `get_vars_from_components` utility to collects the Vars. (But if I'm missing something and you really do need the unused variables, then you will have to go find them with `component_data_objects`. Though it still seems better to find the used ones not on the active sub-tree as well.)

Reply:
I made a simple test and as you said, `component_data_objects` doesn't work if a Var that's used in an active Constraint or Objective is on a deactivated Block. I'll use `get_vars_from_components` here. Thanks for your advice.